app/backend/src/core/scheduler/TaskScheduler.ts

```diff
@@ -15,20 +15,19 @@ export class TaskScheduler {
     this.isTaskRunning = false;
   }
 
-  async start() {
-    if (this.isTaskRunning) {
-      this.logger.debug("Task is already running, skipping this interval.");
-      return;
-    }
-
-    try {
-      this.isTaskRunning = true;
-      await this.executeTask();
-    } catch (error) {
-      this.logger.error("Error in task:", error);
-    } finally {
-      this.isTaskRunning = false;
-    }
+  start() {
+    setImmediate(async () => {
+      if (!this.isTaskRunning) {
```

Comment:
Can we move this block here and in `setInterval` to a separate method to avoid duplication?